lib/rules/base.js

```diff
@@ -30,7 +19,13 @@ module.exports = function(options) {
     this._log = log;
     this.ruleName = ruleName;
     this.severity = defaultSeverity;
+    // To support DOM-scoped configuration instructions, we need to maintain
+    // a stack of our configurations so we can restore the previous one when
+    // the current one goes out of scope. The current one is duplicated in
+    // this.config so the rule implementations don't need to worry about the
+    // fact that there is a stack.
     this.config = this.parseConfig(config);
+    this.configStack = [this.config];
```

Comment:
Do we intend for this to be "public API" for the rules?  I haven't thought deeply about it, but it seems like it might be ok to make it public.  

If we don't want it to be public, we should underscore it.